### DISCUSSION TITLE
Post locking or presence indicator

### DISCUSSION THREAD

**Discussion by jordie23** [keystonejs/keystone#6612] [Event Q&A Q3 2021]
Howdy,

TL;DR: I'm wondering whether the Keystone team has plans on how to solve the issue of two or more people editing the same list item. This is solved on other platforms in different ways including post locking or presence indicators.

More info:

In our organisation it can be common for two or more people to need to edit a piece of content at the same time. At the moment in keystone 6, two or more people editing the same item can lead to changes being lost.

Example scenario using keystone 6 currently:

- Person A starts editing an item
- Person B also starts editing the same item
- Person A saves their changes
- Person B saves their changes
- Person A's changes will be wiped out without knowing. If Person A were to make another edit and save their current state without navigating away, they would then wipe out the changes Person B has made/saved.

This problem is solved in wordpress by "locking" a post when someone is editing. So if Person B came to edit the item in the above scenario, they would see a message and be prevented from editing unless they choose the option to "take over", which would kick Person A out of the post.

Other platforms like Google Docs or Sanity use another method where two people can edit at the same time, and their position in the item editing page is displayed by a presence icon. However this relies on the current state being constantly saved in a draft form and shared amongst all people on that page.

**Comment by gautamsi** [comment]
not in default admin ui, you want to fork the default admin ui or create new form to do this.

**Comment by MurzNN** [comment]
Another solution to content locking is to implement the direct connection between all editors using WebRTC (something like etherpad.org) to sync changes on frontend online (without saving this as draft to backend), but this is much harder to implement.

**Comment by MurzNN** [comment]
And the first easy step can be simply place the "lock" flag and show "This item is now opened for editing by User2, be careful!".

**Comment by MurzNN** [comment]
Also the solution can be storing all revisions of item, with ability to restore previous revision. This feature is implemented in Drupal CMS and works well: when the second user saves the content, after this - first user finishes the editing and overwriting the second's content, both of them are saved as separate revisions, so after this we can view the edition history and restore manually data, that was overwritten.

**Comment by MurzNN** [comment]
And for Content Locking implementation UI good example is this Drupal module: https://www.drupal.org/project/content_lock

**Comment by raveling** [comment]
@jordie23 here's our response in the recent community Q&A event:
https://www.youtube.com/watch?v=r1IJh-iHm1c&t=3041s
🙂
